### ISSUE TITLE
Training parameters and support for other custom parameters/layers/etc.

### ISSUE THREAD

**Issue by ben-arnao** [keras-team/autokeras#1138]
I apologize if i am missing it, but after going through some documentation and the white paper on BO guided network morphism it seems like autokeras has a fixed amount of popular parameters and choices that it optimizes on and chooses from based on what what type of problem we are doing ie. image classification, structured regression. Is this assumption correct? Is there a way to tell ak to include layers like swish activation, gaussian dropout, or are these already an option? What about layer order? ie. batch norm before dropout or after.

Does ak optimize training parameters? How a model is trained is very important to performance as well. I'm not seeing anything in regards to learn rate, optimizer, weight decay, batch size, or optimizing callbacks themselves like reduce lr factor for ReduceLROnPlateau. Is there any attention to learning-related parameters or is this strictly for architecture search?

Lastly, is there an option for overriding and providing custom code for various parts of ak? For example let's say if i want to smooth loss over last few epochs and use this as the model's score instead of just the last epoch. Or if i want to take an average of multiple trials to ensure we are mitigating variance. I'd prefer if there was some method we can override where we are supplied a set of hyperparameters and we just assign a score to it.

Also one last thing... i noticed fit() doesn't take arguments like sample_weight. Not sure if this is known.

I know these may be many different questions but if some of these features aren't included i think it would be helpful to allow users the option. I understand ak is a library more for non-ML users to get "good enough" models, so i understand if these things i'm asking for aren't in the scope of the project, but for more advanced users it would be nice to have these options and still be able to utilize some of the optimization techniques ak has available.
